### ISSUE TITLE
The ToString() method cannot be overridden

### ISSUE THREAD

**Issue by squideyes** [SteveDunn/Vogen#183]
### Describe the bug

When I try to override the ToString method I get a  [CS0121](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0121?f1url=%3FappId%3Droslyn%26k%3Dk(CS0121)) error:

The call is ambiguous between the following methods or properties: 'object.ToString()' and 'object.ToString()'

I'd really like to override the string--either directly or throw an attribute--since the default representation is wrong.

### Steps to reproduce

Create a struct like:

```
[ValueObject(typeof(TimeOnly))]
public partial struct Moment
{
    public override string ToString() => "HH\\:mm\\:ss\\.fff";
}

```
The code will show a squidgly line under "ToString()" and will not compile.

### Expected behaviour

The ToString() method should be able to be overridden

**Comment by SteveDunn** [comment]
Thanks for the report @squideyes !

I'll get around to do this when a PR that is being worked on is merged (unfortunately, because of the snapshot tests that we have, we could end up with thousands of merge conflicts for the verified files). It shouldn't be too long hopefully.

Thanks again!

**Comment by SteveDunn** [comment]
@squideyes , the latest release should address you problem. Let me know if it doesn't. Thanks once again for your feedback!

**Comment by squideyes** [comment]
Thanks for the quick turnaround.

I was able to override ToString(), so that’s great!

I have one (small) nibble, though.  For some reason  when I typed “override” into the text editor, Intellisense didn’t offer up the usual suggestions (i.e. overrides for the Equals, GetHashCode and ToString methods).  I had no problem in typing in the required code (and it did work!), but even so, something weird seems to be going on.

Do keep up the good work!  It’s much appreciated…

**Comment by squideyes** [comment]
I take it back!  

I'm no longer getting the squiggly Intellisense lines but when I try to compile code like the following, I get a "'AccountId' already defines a member called 'ToString' with the same parameter types" error:

```
using System.Text.RegularExpressions;
using Vogen;

namespace SquidEyes.Oanda;

[ValueObject(typeof(string))]
public partial struct AccountId
{
    private static readonly Regex validator =
        new(@"^\d{3}\-\d{3}\-\d{7}\-\d{3}$", RegexOptions.Compiled);

    private static Validation Validate(string value) => CanParse(value)
        ? Validation.Ok
        : Validation.Invalid(GetIsBadValueMessage());

    public static bool CanParse(string value) => 
        validator.IsMatch(value);

    public override string ToString() => Value;

    private static string GetIsBadValueMessage() =>
        $"An AccountId must match the following regex: {validator}.";
}
```

**Comment by SteveDunn** [comment]
Oops, I apologize. I only handle this in classes, and not structs or records. I'll do another build with a proper fix shortly.

But the records case is an interesting one. By default, the compiler auto-generates a `ToString` method. So, when Vogen comes along, it asks 'Is there an overridden `ToString` method?` - and for records, the answer is always **yes**, regardless of whether there is a user supplied one.

**Comment by squideyes** [comment]
I’ve mostly used Vogen for structs, so far, so I don’t see that as much of a problem.

Once you make the change, I’ll be happy to make a pull request that included my samples classes with some included notes, if that would help…

**Comment by SteveDunn** [comment]
Thank you @squideyes - that would certainly help!

I've just checked in a proper fix, and I'm now adding an analyzer that spots the issue. If a record Value Object doesn't seal the ToString ovverride, it generates this compilation error:

`error VOG020: ToString overrides should be sealed on records. See https://github.com/SteveDunn/Vogen/wiki/Records#tostring for more information.`

**Comment by SteveDunn** [comment]
@squideyes - a proper fix is now in 1.0.24, as well as the analyzer that spots non sealed overrides in records.
Thanks for providing your sample instances.  A repository of common 'domain concepts' would be great!  I'll create an Issue to track and discuss it.
